contracts/facets/EmbalmerFacet.sol

```diff
@@ -34,267 +34,237 @@ contract EmbalmerFacet {
+        uint256 maximumRewrapInterval;
+        uint8 threshold;
+        uint256 creationTime;
+    }
 
-    /// @notice Embalmer creates the sarcophagus.
-    ///
-    /// The purpose of createSarcophagus is to:
-    ///   - Lock up payment for the selected archaeologists (digging fees)
-    ///   - Store the arweave TX IDs pertaining to the encrypted file payload
-    ///   -    and the encrypted shards
-    ///   - Verify the selected archaeologists have signed off on the
-    ///         double hash of their key share,
-    ///         arweave tx id storing key shares,
-    ///         and maximumRewrapInterval to be used for lifetime of the sarcophagus
-    ///   - Store the selected archaeologists' addresses, digging fees and
-    ///   -     unencrypted double hashes
-    ///   - Curse each participating archaeologist
-    ///   - Create the sarcophagus object
+    /**
+     * Parameters for an archaeologist's curse, supplied during sarcophagus creation
+     * diggingFee - diggingFee archaeologist has agreed to receive on sarcophagus for its entire lifetime
+     */
+    struct SelectedArchaeologistData {
+        address archAddress;
+        uint256 diggingFee;
+        bytes32 doubleHashedKeyShare;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    /// @notice Creates a sarcophagus with the supplied parameters and locks
+    /// a portion of each archaeologist's freeBond equal to the diggingFees for the sarcophagus.
+    /// Verifies that all supplied archaeologists have signed off on
+    ///    - doubleHashedKeyShare assigned to them
+    ///    - arweaveTxId for encrypted keyshares
+    ///    - maximumRewrapInterval to be enforced for the lifetime of the sarcophagus
+    ///    - creationTime of sarcophagus
+    ///    - diggingFee to be paid to that archaeologist on all rewraps for the lifetime of the sarcophagus
     ///
     /// @param sarcoId the identifier of the sarcophagus
-    /// @param sarcophagus an object that contains the sarcophagus data
+    /// @param sarcophagusParams params to set on sarcophagus being created
     /// @param selectedArchaeologists the archaeologists the embalmer has selected to curse
-    /// @param arweaveTxIds ordered pair of arweave tx ids: [tx storing sarcophagus payload,
-    ///         tx storing the archaeologists' encrypted key shares]
-    /// @return The index of the new sarcophagus
+    /// @param arweaveTxIds ordered pair of arweave tx ids: [sarcophagus payload tx, encrypted key share tx]
+    /// @return The index of the new sarcophagus in sarcophagusIdentifiers
     function createSarcophagus(
         bytes32 sarcoId,
-        LibTypes.SarcophagusMemory memory sarcophagus,
-        LibTypes.SelectedArchaeologistData[] memory selectedArchaeologists,
-        string[] memory arweaveTxIds
+        SarcophagusParams calldata sarcophagusParams,
+        SelectedArchaeologistData[] calldata selectedArchaeologists,
+        string[2] memory arweaveTxIds
```

Comment:
Typing this as a tuple enforces the validation we were doing to be sure that 2 ids are passed in